Quadriga/src/main/java/edu/asu/spring/quadriga/service/passthroughproject/IPassThroughProjectManager.java

```diff
@@ -0,0 +1,59 @@
+package edu.asu.spring.quadriga.service.passthroughproject;
+
+import javax.xml.bind.JAXBException;
+
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.domain.impl.passthroughproject.PassThroughProject;
+import edu.asu.spring.quadriga.domain.impl.passthroughproject.PassThroughProjectInfo;
+import edu.asu.spring.quadriga.domain.passthroughproject.IPassThroughProject;
+import edu.asu.spring.quadriga.exceptions.NoSuchRoleException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+
+public interface IPassThroughProjectManager {
+
+    /**
+     * This method is used to save the {@link PassThroughProject} object in the
+     * database.
+     * 
+     * @param user
+     *            The user info object.
+     * @param project
+     *            The pass through info object
+     * @return The internal project id
+     * @throws QuadrigaStorageException
+     * @throws NoSuchRoleException
+     */
+    String savePassThroughProject(IUser user, IPassThroughProject project)
+            throws QuadrigaStorageException, NoSuchRoleException;
+
+    /**
+     * This method will return the workspace id for an external project. If the
+     * provided id does not exist in the database a new workspace entry will be
+     * created and its id will be returned.
+     * 
+     * @param passThroughProjectInfo
+     *            The {@link PassThroughProjectInfo} object.
+     * @param projectId
+     *            The project id of the project.
+     * @param user
+     *            The user info object.
+     * @return The workspace id for an external project.
+     * @throws JAXBException
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
```

Comment:
I'm confused about what this method does. What does "This method will return the workspace id for an external project" mean? There can be multiple workspaces in a project, which workspace id will the method return? And what does it mean that "if the provided id does not exist in the database"? The project id? Why would the method create a new workspace if there is no project (instead of creating a project)? Also, why is the project id not part of the passthroughprojectinfo?